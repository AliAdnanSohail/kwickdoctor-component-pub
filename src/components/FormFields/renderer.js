import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
