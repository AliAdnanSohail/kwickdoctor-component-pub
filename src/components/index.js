export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as FileViewer } from './FileViewer';
export { default as LinkButton } from './LinkButton';
export { default as Loader } from './Loader';
export { default as Modal } from './Modal';
export { default as Notification } from './Notification';
export { default as Pagination } from './Pagination';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';
export { default as VideoPlayer } from './VideoPlayer';
export { default as VideoRecorder } from './VideoRecorder';

export * from './FormFields';
export * from './Navigation';
export * from './Stepper';
