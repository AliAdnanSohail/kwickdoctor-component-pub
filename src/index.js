export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
