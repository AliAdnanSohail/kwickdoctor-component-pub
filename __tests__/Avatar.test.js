import React from 'react';
import renderer from 'react-test-renderer';

import { Avatar } from '../src';

it('renders correctly', () => {
  const element = renderer.create(<Avatar />).toJSON();

  expect(element).toMatchSnapshot();
});
