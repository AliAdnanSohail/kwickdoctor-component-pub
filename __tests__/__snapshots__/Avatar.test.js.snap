// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="jsx-4055291026 avatar"
  style={
    Object {
      "backgroundImage": "",
      "borderRadius": "3px",
      "height": "40px",
      "width": "40px",
    }
  }
>
  <i
    className="jsx-4055291026 material-icons"
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    person_outline
  </i>
</div>
`;
